dxcint/tests/test_Messenger.py

```diff
@@ -1,9 +1,45 @@
 import pytest
-from dxcint.Messenger import Messenger
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dxcint.RegisteredTest import RegisteredTest
+from copy import copy
+import time
+from dxcint.Messenger import State
 
 
-@pytest.fixture(scope="session")
-def messenger_init():
-    yield Messenger()
+def track_history(messenger):
+    timeout = 3600
+    t = 0
+    history = []
+    while t < timeout:
+        current_state = messenger.query_state()
+        history.append(current_state)
+        if current_state == State.FINISHED or current_state == State.FAIL:
+            break
+        time.sleep(15)
+        t += 15
+    return history
 
-# TODO how to poll messages
+
+@pytest.mark.usefixtures("build_executable_wdl")
+def test_parallel_execution(build_executable_wdl: RegisteredTest):
+    test1 = build_executable_wdl
+    test2 = copy(build_executable_wdl)
+    messenger1 = test1.messenger
+    messenger2 = test2.messenger
+    results = []
+    with ThreadPoolExecutor(max_workers=10) as executor:
+        h1 = executor.submit(track_history, messenger1)
+        h2 = executor.submit(track_history, messenger2)
+        test_futures = {
+            executor.submit(test1.get_test_result),
+            executor.submit(test2.get_test_result)
+        }
+        results = [f.result() for f in as_completed(test_futures)]
+        history1 = h1.result()
+        history2 = h2.result()
+
+    assert results == [False, False]
+    assert State.NOT_DONE in history1
```

Comment:
Can you test for the order at which those statuses appear to confirm concurrency? For example - create a dict with both histories and update it with the status changes. Then check that test 1 did not wait for test 2 to finish

Reply:
Added a simple check for that the jobs share some time in NOT_DONE state but beyond that the workers are quite unpredictable, so one waiting and one being done is not impossible even if they are submitted at the same time.